ESP32C3 failes to build with CONFIG_RTC_CLK_SRC_EXT_CRYS=y

Using zephyr main, you get build error when enabling external 32K clock on esp32c3.
To enable external 32K clock I use the Kconfig
`CONFIG_RTC_CLK_SRC_EXT_CRYS=y`

Repo:
Fetch zephyr according to Getting started guide.

Modifiy hello world sample by adding the line
`CONFIG_RTC_CLK_SRC_EXT_CRYS=y`
to prj.conf

Build hello world using the line 
`west build -p -b esp32c3_devkitm samples/hello_world`

The following  build error now happens:
/home/fgihl/tmp/zephyr-validation/modules/hal/espressif/components/bootloader_support/src/bootloader_clock_init.c: In function 'bootloader_clock_configure':
/home/fgihl/tmp/zephyr-validation/modules/hal/espressif/components/bootloader_support/src/bootloader_clock_init.c:85:31: error: 'CONFIG_ESP_SYSTEM_RTC_EXT_XTAL_BOOTSTRAP_CYCLES' undeclared (first use in this function)
   85 |         rtc_clk_32k_bootstrap(CONFIG_ESP_SYSTEM_RTC_EXT_XTAL_BOOTSTRAP_CYCLES);
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/fgihl/tmp/zephyr-validation/modules/hal/espressif/components/bootloader_support/src/bootloader_clock_init.c:85:31: note: each undeclared identifier is reported only once for each function it appears in
[126/198] Building C object zephyr/arch/arch/riscv/core/CMakeFiles/arch__riscv__core.dir/fatal.c.obj
ninja: build stopped: subcommand failed.

--- a/soc/espressif/esp32/Kconfig.rtc
+++ b/soc/espressif/esp32/Kconfig.rtc
@@ -74,4 +74,21 @@ config RTC_XTAL_CAL_RETRY
 	  Increase this option if the 32k crystal oscillator
 	  does not start and switches to internal RC.
 
+config ESP_SYSTEM_RTC_EXT_XTAL_BOOTSTRAP_CYCLES
+	int "Bootstrap cycles for external 32kHz crystal"
+	depends on ESP_SYSTEM_RTC_EXT_XTAL
+	default 5
+	range 0 32768
+	help
+	  To reduce the startup time of an external RTC crystal,
+	  we bootstrap it with a 32kHz square wave for a fixed number of cycles.
+	  Setting 0 will disable bootstrapping (if disabled, the crystal may take
+	  longer to start up or fail to oscillate under some conditions).
+
+	  If this value is too high, a faulty crystal may initially start and then fail.
+	  If this value is too low, an otherwise good crystal may not start.
+
+	  To accurately determine if the crystal has started,
+	  set a larger "Number of cycles for RTC_SLOW_CLK calibration" (about 3000).
+
 endif # SOC_SERIES_ESP32

--- a/soc/espressif/esp32c3/Kconfig.rtc
+++ b/soc/espressif/esp32c3/Kconfig.rtc
@@ -47,4 +47,21 @@ config RTC_CLK_CAL_CYCLES
 	    In case more value will help improve the definition of the launch of the crystal.
 	    If the crystal could not start, it will be switched to internal RC.
 
+config ESP_SYSTEM_RTC_EXT_XTAL_BOOTSTRAP_CYCLES
+	int "Bootstrap cycles for external 32kHz crystal"
+	depends on ESP_SYSTEM_RTC_EXT_XTAL
+	default 0
+	range 0 32768
+	help
+	  To reduce the startup time of an external RTC crystal,
+	  we bootstrap it with a 32kHz square wave for a fixed number of cycles.
+	  Setting 0 will disable bootstrapping (if disabled, the crystal may take
+	  longer to start up or fail to oscillate under some conditions).
+
+	  If this value is too high, a faulty crystal may initially start and then fail.
+	  If this value is too low, an otherwise good crystal may not start.
+
+	  To accurately determine if the crystal has started,
+	  set a larger "Number of cycles for RTC_SLOW_CLK calibration" (about 3000).
+
 endif # SOC_SERIES_ESP32C3

--- a/soc/espressif/esp32s2/Kconfig.rtc
+++ b/soc/espressif/esp32s2/Kconfig.rtc
@@ -73,4 +73,21 @@ config RTC_XTAL_CAL_RETRY
 	  Increase this option if the 32k crystal oscillator
 	  does not start and switches to internal RC.
 
+config ESP_SYSTEM_RTC_EXT_XTAL_BOOTSTRAP_CYCLES
+	int "Bootstrap cycles for external 32kHz crystal"
+	depends on ESP_SYSTEM_RTC_EXT_XTAL
+	default 0
+	range 0 32768
+	help
+	  To reduce the startup time of an external RTC crystal,
+	  we bootstrap it with a 32kHz square wave for a fixed number of cycles.
+	  Setting 0 will disable bootstrapping (if disabled, the crystal may take
+	  longer to start up or fail to oscillate under some conditions).
+
+	  If this value is too high, a faulty crystal may initially start and then fail.
+	  If this value is too low, an otherwise good crystal may not start.
+
+	  To accurately determine if the crystal has started,
+	  set a larger "Number of cycles for RTC_SLOW_CLK calibration" (about 3000).
+
 endif # SOC_SERIES_ESP32S2

--- a/soc/espressif/esp32s3/Kconfig.rtc
+++ b/soc/espressif/esp32s3/Kconfig.rtc
@@ -43,4 +43,21 @@ config RTC_CLK_CAL_CYCLES
 	  In case more value will help improve the definition of the launch of the crystal.
 	  If the crystal could not start, it will be switched to internal RC.
 
+config ESP_SYSTEM_RTC_EXT_XTAL_BOOTSTRAP_CYCLES
+	int "Bootstrap cycles for external 32kHz crystal"
+	depends on ESP_SYSTEM_RTC_EXT_XTAL
+	default 0
+	range 0 32768
+	help
+	  To reduce the startup time of an external RTC crystal,
+	  we bootstrap it with a 32kHz square wave for a fixed number of cycles.
+	  Setting 0 will disable bootstrapping (if disabled, the crystal may take
+	  longer to start up or fail to oscillate under some conditions).
+
+	  If this value is too high, a faulty crystal may initially start and then fail.
+	  If this value is too low, an otherwise good crystal may not start.
+
+	  To accurately determine if the crystal has started,
+	  set a larger "Number of cycles for RTC_SLOW_CLK calibration" (about 3000).
+
 endif # SOC_SERIES_ESP32S3

